### DISCUSSION TITLE
What happened to QueryState ?

### DISCUSSION THREAD

**Discussion by callmephil** [zino-app/graphql-flutter#1248] [Q&A]
Hello, 

I'm trying to upgrade from 5.0.1. But I can't since QueryState has been removed.

Previously I used GlobalKey<QueryState> in my parent container to refetch multiple Query at once, from different sub-widgets.

Sample:
```
  final myVarKey = GlobalKey<QueryState>();
  
  // Refetch func 
  await myVarKey.currentState?.observableQuery!.refetch();
  
  // Build method
  ChildWidget(refetchKey: myVarKey)
  
  // Child Widget (File 2)
  Query(
  key: widget.refetchKey,
  options...
  builder...
  ),
```

But since the migration to hooks, I can't seem to find an alternative. Has anyone experienced this or has an alternative for triggering refetch from a sub-widget?

**Comment by vincenzopalazzo** [comment]
I suggest to you update to the last beta version, and after that post your code example (minimal and completed), 5.0.1 is almost 2 years old now

**Reply by callmephil** [reply]
That's the thing, I'm upgrading to the main channel. But I can't find a replacement for QueryState.
In many parts of my application, I use the RefreshIndicator widget, which sometimes is above or outside the Query widget.
How can I access the refetch method outside the builder in the newer version of graphql_flutter?

**Reply by vincenzopalazzo** [reply]
I do not think there is anymore query state, try to look inside the new state https://github.com/zino-hofmann/graphql-flutter/tree/main/packages/graphql_flutter#query and model the current library to what you are doing

I think you need the Query Widget, but not sure I'm not familiar with the QueryState concept
